### PR TITLE
Handle post-install-script for provisioning openstack instance

### PR DESCRIPTION
**Pull request by skycastlelily**
Pull Request Checklist

* [ ] implement the feature
* [ ] write the documentation
* [ ] extend the test coverage
* [ ] update the specification
* [ ] adjust plugin docstring
* [ ] modify the json schema
* [ ] mention the version
* [ ] include a release note

### CODE REVIEW

**Comment by happz** [comment]
One suggestion so far, I'd recommend changing the $SUBJ. The patch does not add the feature, the post-install script was already supported, even for OpenStack provisioning. What it adds is support for the post-install script being a URL, i.e. the plugin would download it first, then it would be passed down to Artemis. And that is also not something specific for OpenStack, it would have benefits for AWS or Beaker provisioning as well.

File: tmt/steps/provision/artemis.py [hunk 1]
```diff
@@ -548,7 +548,22 @@ def _create(self) -> None:
         # TODO: spot instance
 
         if self.post_install_script:
-            data['post_install_script'] = self.post_install_script
+            if tmt.utils.is_url(self.post_install_script):
+                try:
+                    response = requests.get(self.post_install_script)
```
Comment (lukaszachy) [inline tmt/steps/provision/artemis.py:553]: Please use retry_session, for inspiration: https://github.com/teemtee/tmt/blob/1b6ac468c7ab26ca43b7c9ed006a48ce4e3dec01/tmt/steps/prepare/ansible.py#L127

File: tmt/steps/provision/artemis.py [hunk 2]
```diff
@@ -548,7 +548,22 @@ def _create(self) -> None:
...
+                    response.raise_for_status()
+
+                except requests.exceptions.RequestException:
+                    raise ArtemisProvisionError(
+                        "failed to get the post_install_script file")
+
+                post_install_script = response.text
+            else:
+                self.info('post-install-script argument is treated as a raw script')
```
Comment (lukaszachy) [inline tmt/steps/provision/artemis.py:562]: Is this worth of 'info' level? Isn't it more like debug?

**Review by lukaszachy — CHANGES_REQUESTED** [review]

File: tmt/steps/provision/artemis.py (hunk 2 above)
Comment (skycastlelily) [reply tmt/steps/provision/artemis.py]: @lukaszachy Thanks for your review,updated accordingly:)

File: tmt/steps/provision/artemis.py [hunk 3]
```diff
@@ -548,7 +548,26 @@ def _create(self) -> None:
...
         if self.post_install_script:
-            data['post_install_script'] = self.post_install_script
+            if tmt.utils.is_url(self.post_install_script):
+                try:
+                    with retry_session() as session:
+                        response = session.get(self.post_install_script)
+
+                    if not response.ok:
+                        raise ArtemisProvisionError(
+                            "Failed to get the post_install_script file")
```
Comment (lukaszachy) [inline tmt/steps/provision/artemis.py:558]: How about 'download' instead of get?

**Review by lukaszachy — CHANGES_REQUESTED** [review]
Thank you for the update, could you edit the the --help message as well? Currently it is not mentioned that SCRIPT can be remote location

```
--post-install-script SCRIPT If set, this script will be executed on the 
      guest after provisioning.
```

File: tmt/steps/provision/artemis.py (hunk 3 above)
Comment (happz) [reply tmt/steps/provision/artemis.py:558]: And let's be more civil and consistent too: there is no such thing as "post_install_script" in what the user can see. They can see a CLI option, `--post-install-script`, or an fmf key, `post-install-script`; a name with underscores is a leak of an internal variable or key which muddies the waters. How about "Failed to download the post-install script."?

Comment (happz) [reply tmt/steps/provision/artemis.py:558]: The same happens on line 562, BTW.

**Review by lukaszachy — APPROVED** [review]
LGTM, thank you.

**Review by happz — APPROVED** [review]

**Pull request merged**